lizmap/modules/lizmap/controllers/edition.classic.php

```diff
@@ -398,9 +398,9 @@ public function createFeature()
             return $this->serviceAnswer();
         }
 
-        jForms::destroy('view~edition', $this->featureId);
+        jForms::destroy('view~edition', $this->layerId.'||'.$this->featureId);
```

Comment:
```suggestion
        jForms::destroy('view~edition', $this->formId());
```

Après avoir défini le méthode `formId`
```php
protected function formId() {
    if ($this->featureId) {
        return $this->layerId.'||'.'||'.$this->featureId;
    }
    return $this->layerId; // ou $this->layerId.'||____new__feature___';
}
```

Reply:
Good catch. I also thought about creating a unique form ID for each "new feature". So that each form would be unique. We would also need to have different form node ID property for each form.

Reply:
@laurentj In the context of object creation (parent and optional children) what about using a parameter UID filled by the JS code when calling the `edition` methods and use it to generate the form ID ? I would use a UUID such as `96944c88-b947-4b69-a538-d9605193bd57` (or a simpler hash).

The JS code would then keep the form's uniques IDs and be able to manage validations for each form (and hide/show the forms, for example if we show a table of children data "to be added" with buttons to modify each child form data.

We could have a IRL conversation about it ;)

Reply:
Unfortunately, in the case of there multiple forms of the same type in the same HTML page, even if they have a different form ID, the HTML and JS code cannot work correctly. All HTML field have the same html id across all forms, so a `document.getElementById()` won't work. 

I must study how we can solve this issue. A solution is to change how the html id are built, but I'm afraid that it will brake all existing custom JS scripts that rely on these html id... And this change cannot be made in the current stable version of Jelix, so it will be made into Jelix 1.9, so for Lizmap 3.8 or 4.0..

I'll try to find another solution...